v6/v6_interface.ml

```diff
@@ -92,28 +104,37 @@ module API(R : RPC) = struct
   (* define global parameters for API calls *)
   let debug_info_p = Param.mk ~description:[
     "An uninterpreted string to associate with the operation."
-    ] Types.string 
+    ] debug_info 
+
+  (* ---- TODO : include functor to give 'err' type : TODO ---- *)
+
+  module E  = Idl.Error.Make(struct type t = LicensingError let t=LicensingError end)
+
+  let err = Error.{
+    def = errors;
+    raiser = (function | e -> raise (LicensingError e));
+    matcher = function | LicensingError e -> Some e | _ -> None
+  }
 
   (* dbg_str -> requested edition -> current params -> edition_info *)
   let apply_edition =
-    declare
-    "apply_edition"
+    let edition_p = Param.mk ~description:["Edition tite"] Types.string in
+    let edition_info_p = Param.mk ~description:["Edition info"] edition_info in 
+    let current_params_p = Param.mk ~description:["Xapi paramaters"] stringPairLst in
```

Comment:
"parameters"

Reply:
oh god I fixed that in one version and must have lost it among the many rebases. I'll go hang my head in shame now